src/hybrid_reader.c

```diff
@@ -210,6 +215,30 @@ static void prepareResults(HybridIterator *hr) {
     if (heap_count(hr->topResults) == hr->query.k) {
       break;
     }
+    // Re-evaluate the child num estimated results and the hybrid policy based on the current batch.
+    size_t new_results_cur_batch = heap_count(hr->topResults) - (hr->query.k - n_res_left);
+    // This is the ratio between index_size to child results size as reflected by this batch.
+    float cur_ratio = (float)new_results_cur_batch / n_res_left;
+    // Child estimated number of results as reflected by this batch.
+    size_t cur_child_num_estimated = cur_ratio * VecSimIndex_IndexSize(hr->index);
+    // Conclude the new estimation of the child res num as the average between the old
+    // and new estimation (get the accumulated estimation).
+    child_num_estimated = (child_num_estimated + cur_child_num_estimated) / 2;
```

Comment:
Why average? Especially when based on the upper bound of NumEstimated() which can be completely off.

Reply:
Because I don't want to to "tikun tothanim" also. Perhaps I got 0 results in the batch by chance, even though `NumEstimated()` was accurate. In that case I don't mistakenly think that `child_num_estimated = 0` and change policy accordingly.